src/main/java/org/alfresco/repo/search/impl/parsers/FTS.g

```diff
@@ -772,18 +799,27 @@ range_right
                 -> EXCLUSIVE
         ;
 
-/* Need to fix the generated parser for extra COLON check ??*/
-
 fieldReference
+        :
+        fieldReferenceWithPrefix
+        | fieldReferenceWithoutPrefix;
+
+fieldReferenceWithoutPrefix
         :
         AT?
-        (
-                  (prefix) => prefix
-                | uri
-        )?
+        uri?
         identifier
                 ->
-                        ^(FIELD_REF identifier prefix? uri?)
+                        ^(FIELD_REF identifier uri?)
+        ;
+
+fieldReferenceWithPrefix
+        :
+        AT?
+        (prefix) => prefix

```

Comment:
I think we can avoid (prefix) => prefix because there are no other alternatives in this rule. We can write directly:

        AT?
        prefix  identifier
                ->
                        ^(FIELD_REF identifier prefix)
     ;

Reply:
Done